app/models/user.rb

```diff
@@ -24,18 +24,44 @@ def sug_users
     users_array.compact
   end
 
-  # Users who have yet to confirme friend requests
-  def pending_friends
-    friendships.map { |friendship| friendship.friend if friendship.confirmed == false }.compact
+  def friends_posts_and_own_posts
+    posts_array = posts.map { |post| post if post.user == self }
+    self.friends.each do |f|
+      posts_array += f.posts
+    end
+    posts_array.compact
+    posts_array.sort_by(&:created_at).reverse
   end
 
-  # Users who have requested to be friends
-  def friend_requests
-    inverse_friendships.map { |friendship| friendship.user if friendship.confirmed == false }.compact
+  def show_pending_friends
+    pending_f_array = friendships.map { |friendship|
+      friendship.friend if friendship.confirmed == false and friendship.user == self
+    }
+    pending_f_array.compact
+  end
+
+  def show_friendships(pending_user)
+    friendships_array = friendships.map { |f| f if f.user == self and f.friend == pending_user }
+    friendships_array.compact
+  end
+
+  def show_friend_requests
+    f_r_array = inverse_friendships.map { |friendship| friendship.user if friendship.confirmed == false and friendship.friend == self }
+    f_r_array.compact
+  end
+
+  def show_friendship_requests(user)
+    fsp_array = inverse_friendships.map { |f| f if f.user == user and f.friend == self }
+    fsp_array.compact
+  end
+
+  def confirm_friendship(user)
+    confirm_f_array = inverse_friendships.map { |f| f if f.user == user and f.friend == self }
+    confirm_f_array.compact
   end
```

Comment:
**OPTIONAL**
We can use the Active Record interface to query information instead of ruby methods. Active Record methods are optimized to query information and do that in a faster way then ruby methods. Check out this [link](https://guides.rubyonrails.org/active_record_querying.html) to learn more about it

Reply:
Thanks,
Good to know that, in this project, I have used ruby methods from the previous milestone taking  into account the suggested article: https://smartfunnycool.com/friendships-in-activerecord/
In this milestone, I learned how to create my own methods to achieve the desired results in this project, in future projects I will use Active Record to query information in a faster way.